<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1293</width>
    <height>1049</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>691</width>
      <height>761</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="imagesLayout">
     <property name="spacing">
      <number>5</number>
     </property>
     <property name="leftMargin">
      <number>5</number>
     </property>
     <property name="topMargin">
      <number>5</number>
     </property>
     <property name="rightMargin">
      <number>5</number>
     </property>
     <property name="bottomMargin">
      <number>5</number>
     </property>
     <item>
      <widget class="QLabel" name="modifiedImage">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="originalImage">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>20</y>
      <width>521</width>
      <height>138</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="imageOperationsLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Select image:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="loadFileButton">
       <property name="text">
        <string>Load</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="saveFileButton">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="expandImageButton">
       <property name="text">
        <string>Expand</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>360</y>
      <width>511</width>
      <height>198</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="convolutionFiltersLayout">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Convolution filters:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="gaussianBlurButton">
       <property name="text">
        <string>Gaussian Blur</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="meanBlurButton">
       <property name="text">
        <string>Mean Blur</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="labBlurButton">
       <property name="text">
        <string>Edge Preserving Blur</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="edgeDetectionButton">
       <property name="text">
        <string>Edge Detection</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sharpenButton">
       <property name="text">
        <string>Sharpen</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="embossButton">
       <property name="text">
        <string>Emboss</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>160</y>
      <width>511</width>
      <height>181</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="functionalFiltersLayout">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Function filters:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="inversionButton">
       <property name="text">
        <string>Inversion</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="brightnessButton">
       <property name="text">
        <string>Brightness Correction</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="contrastButton">
       <property name="text">
        <string>Contrast Enhancement</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="gammaButton">
       <property name="text">
        <string>Gamma Correction</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>590</y>
      <width>177</width>
      <height>390</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="kernelEditorLayout">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Select filter size</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="kernelXSpinBox">
       <property name="minimum">
        <number>3</number>
       </property>
       <property name="maximum">
        <number>9</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="kernelYSpinBox">
       <property name="minimum">
        <number>3</number>
       </property>
       <property name="maximum">
        <number>9</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Select anchor:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="anchorXSpinBox">
       <property name="maximum">
        <number>8</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="anchorYSpinBox">
       <property name="maximum">
        <number>8</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Select weight and bias</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="weightSpinBox"/>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="biasSpinBox">
       <property name="minimum">
        <double>-99.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="computeWeightButton">
       <property name="text">
        <string>Compute weight</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="loadFilterButton">
       <property name="text">
        <string>Load filter</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="saveFilterButton">
       <property name="text">
        <string>Save filter</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="applyCustomConvolution">
       <property name="text">
        <string>Apply filter</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>590</y>
      <width>331</width>
      <height>291</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="kernelGridLayout"/>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1293</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
